### PR TITLE
fixed navi bugs

### PR DESCRIPTION
**Pull request by zstreet87**
* fixed 32 threads per warp
 * set's ROCM_PATH in mlir front-end
 * relaxes auto_mixed_percision_test tol to pass on navi

### CODE REVIEW

File: tensorflow/compiler/mlir/runlit.cfg.py [hunk 1]
```diff
@@ -63,7 +63,7 @@
 # Tweak the PATH to include the tools dir.
 llvm_config.with_environment('PATH', config.llvm_tools_dir, append_path=True)
 
-for key in ['HIP_VISIBLE_DEVICES', 'CUDA_VISIBLE_DEVICES',
+for key in ['ROCM_PATH', 'HIP_VISIBLE_DEVICES', 'CUDA_VISIBLE_DEVICES',
```
Comment (i-chaochen) [inline tensorflow/compiler/mlir/runlit.cfg.py:66]: may I ask why this `ROCM_PATH` can do this magic? somehow navi31 cannot load ROCm in mlir front-end?

**Review by i-chaochen — APPROVED** [review]

File: tensorflow/compiler/mlir/runlit.cfg.py (hunk 1 above)
Comment (zstreet87) [reply tensorflow/compiler/mlir/runlit.cfg.py:66]: This was entirely my setup and shouldn't have been seen anywhere else, sorry! However, the ROCM_PATH wasn't set so this makes things consistent with what the code is expecting at least.

See this code block: https://github.com/ROCm/tensorflow-upstream/blob/develop-upstream/third_party/xla/xla/service/gpu/llvm_gpu_backend/gpu_backend_lib.cc#L773-L779

**Review by jayfurmanek — APPROVED** [review]

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please

**Review by pemeliya — APPROVED** [review]

**Comment by i-chaochen** [comment]
retest Ubuntu-GPU-single please

File: tensorflow/core/kernels/reduction_gpu_kernels.cu.h [hunk 2]
```diff
@@ -736,7 +736,7 @@ void LaunchScalarReduction(OpKernelContext* ctx, OUT_T out, IN_T in,
                            int in_size, Op op, T init,
                            const gpuStream_t& cu_stream) {
 #if TENSORFLOW_USE_ROCM
-  int WARPSIZE = isGfx10(ctx) ? 32 : 64;
+  int WARPSIZE = isGfx10orGfx11(ctx) ? 32 : 64;
```
Comment (draganmladjenovic) [inline tensorflow/core/kernels/reduction_gpu_kernels.cu.h:739]: Don't do this! hipDeviceProp_t has warp size field. https://rocm.docs.amd.com/projects/HIP/en/latest/doxygen/html/structhip_device_prop__t.html#af3357d33c004608bf05bc21a352be81b

Comment (i-chaochen) [reply tensorflow/core/kernels/reduction_gpu_kernels.cu.h:739]: found the source of evil....   :D  https://github.com/ROCm/tensorflow-upstream/commit/5b434ff3c7de26c1dbf5ce9eb2c87f28e1548532

**Pull request merged**